core/src/style/polylineStyle.cpp

```diff
@@ -10,16 +10,36 @@
 #include "tile/tile.h"
 #include "util/mapProjection.h"
 #include "util/extrude.h"
+
 #include "glm/vec3.hpp"
+#include "glm/gtc/type_precision.hpp"
+
+#define EXTRUSION_SCALE 4096.0f
+#define POSITION_SCALE 4096.0f
+#define TEXTURE_SCALE 16384.0f
+#define ORDER_SCALE 2.0f
```

Comment:
We put the outline between the usual integer orders: 

    float outlineOrder = std::min(params.outlineOrder, params.order) - 0.5f;

So to keep the modified order, when packed as integer, it needs to be scaled up - and down again in the shader.
 I'm not completely sure about this whole logic either